### PR TITLE
Backport #457 to 2.5.x

### PR DESCRIPTION
**Pull request by shaobo-he-aws**
## Description of changes

#457 contains the fixes of #456 for release 2.4.

## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [x] A bug fix or other functionality change requiring a patch to `cedar-policy`.

I confirm that this PR (choose one, and delete the other options):

- [x] Updates the "Unreleased" section of the CHANGELOG with a description of my change (required for major/minor version bumps).

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [x] Does not require updates because my change does not impact the Cedar formal model or DRT infrastructure.

## Disclaimer

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cedar-policy/CHANGELOG.md [hunk 1]
```diff
@@ -57,6 +57,7 @@ exist to the `RecordAttrDoesNotExist` error message.
 Now it will not, since we know `{"foo": 5} has bar` is `False`, and the
 validator will return an error for a policy that can never fire.
 - Removed deprecated `__expr` escapes from integration tests.
+- Calling `add_template` with a `PolicyId` that is an existing link will now error.
```
Comment (aaronjeline) [inline cedar-policy/CHANGELOG.md:60]: It's more than just `add_template` right? Also `add` and `link`?

**Review by aaronjeline — APPROVED** [review]

File: cedar-policy/CHANGELOG.md (hunk 1 above)
Comment (shaobo-he-aws) [reply cedar-policy/CHANGELOG.md:60]: I think this PR just fixes the function. The comments of `add` and `link` probably are added to be more informative?

Comment (john-h-kastner-aws) [reply cedar-policy/CHANGELOG.md:60]: I agree. this only changes `add_template`.

**Review by john-h-kastner-aws — APPROVED** [review]

**Comment by khieta** [comment]
I assume the PR title should say 2.4.x instead of 2.5.x?

**Pull request merged**